internal/pkg/bashexec/bashexec.go

```diff
@@ -0,0 +1,60 @@
+package bashexec
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// BashRunner is a wrapper for executing in-memory bash scripts
+type BashRunner struct {
+	name string
+	cmd  *exec.Cmd
+}
+
+// NewBashRunner creates a bash executor from in-memory shell script.
+func NewBashRunner(name, src string, args ...string) (*BashRunner, error) {
+	f, err := os.CreateTemp("", name)
+	if err != nil {
+		return nil, err
+	}
+	if _, err := io.Copy(f, strings.NewReader(src)); err != nil {
+		return nil, err
+	}
+	if err := os.Remove(f.Name()); err != nil {
+		return nil, err
+	}
+
+	fullargs := append([]string{"-c", ". /proc/self/fd/3", name}, args...)
```

Comment:
so `fd/1` and `fd/2` are `stdout`/`stderr` and we know using `fd/3` is safe because we just opened a new file with `os.CreateTemp` above. Maybe we add a comment above the `CreateTemp` that says something like "This will be file descriptor 3 (`fd/3`)".